src/state/slices/txHistorySlice/utils.ts

```diff
@@ -58,17 +58,30 @@ export const deserializeTxIndex = (txIndex: TxIndex): TxDescriptor => {
   // Split the serialized index back into its components
   const parts = txIndex.split(UNIQUE_TX_ID_DELIMITER)
 
+  const accountId = parts[0]
+  const txid = parts[1]
+  const pubkey = parts[2]
+
+  if (!(accountId && txid && pubkey)) {
+    throw new Error(`Invalid tx index: ${txIndex}`)
+  }
```

Comment:
Technically a breaking change in a way, a non-breaking one would be to have non-nullish coallescing to `''`,  though if some or all were ever undefined, we would upsert Txs with parts being undefined in the store, and then, problems.

Reply:
Heh, breaking as in handling the error in the correct place 🧠